### PR TITLE
[DESK-1193] rest api changes

### PR DESCRIPTION
**Pull request by sonali-remote**
### Checklist

- [ ] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)
- [ ] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [ ] I have added one or more reviewers
- [ ] I have added/updated tests for any code changes
- [ ] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [ ] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] Translation strings added/updated for all user-visible text
- [ ] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [ ] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [ ] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Sign in
2. Go to...
3. Click...
4. You should see X, Y, Z...

### Ticket

<https://remoteit.atlassian.net/browse/DESK-1193>

### CODE REVIEW

File: backend/src/User.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import debug from 'debug'
 import Logger from './Logger'
 import EventBus from './EventBus'
 import path from 'path'
+import axios from 'axios'
 import { r3 } from './remote.it'
```
Comment (JamieRuderman) [inline backend/src/User.ts:9]:
```suggestion
```

Please completely remove the r3 dependency and module from the backend codebase and package.json. You will have to move over or re-write the logic in the backend user class.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]

File: backend/src/User.ts [hunk 2]
```diff
@@ -5,7 +5,8 @@ import debug from 'debug'
 import Logger from './Logger'
 import EventBus from './EventBus'
 import path from 'path'
-import { r3 } from './remote.it'
+import axios from 'axios'
+// import { r3 } from './remote.it'
```
Comment (JamieRuderman) [inline backend/src/User.ts:9]: This can be removed

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Looks great @sonali-remote ! Just one little code style note please, then let's merge this!

File: backend/src/User.ts (hunk 1 above)
Comment (sonali-remote) [reply backend/src/User.ts]: done

File: backend/src/User.ts [hunk 3]
```diff
@@ -47,7 +49,11 @@ export class User {
     Logger.info('Attempting auth hash login', { username: credentials.username })
 
     try {
-      const user = await r3.user.authHashLogin(credentials.username, credentials.authHash)
+      const userName = credentials.username
+      const authHash = credentials.authHash
+      
+      const user = await axios.post('/user/login/authhash', { userName, authHash })
+                              .then(resp => this.process(resp, authHash))
```
[the remark this answers is not in the document]
Comment (sonali-remote) [reply backend/src/User.ts]: done

File: backend/src/User.ts (hunk 2 above)
Comment (sonali-remote) [reply backend/src/User.ts]: done

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
@sonali-remote I went to test the code and it failed with this error:

```
[watch-electron   ] LOGIN AUTH FAILURE warn 2022-06-28T22:50:28.966Z {
[watch-electron   ]     "username": "jamie@remote.it",
[watch-electron   ]     "error": {
[watch-electron   ]         "message": "connect ECONNREFUSED 127.0.0.1:80",
[watch-electron   ]         "name": "Error",
[watch-electron   ]         "stack": "Error: connect ECONNREFUSED 127.0.0.1:80\n    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)",
[watch-electron   ]         "config": {
[watch-electron   ]             "transitional": {
[watch-electron   ]                 "silentJSONParsing": true,
[watch-electron   ]                 "forcedJSONParsing": true,
[watch-electron   ]                 "clarifyTimeoutError": false
[watch-electron   ]             },
[watch-electron   ]             "transformRequest": [
[watch-electron   ]                 null
[watch-electron   ]             ],
[watch-electron   ]             "transformResponse": [
[watch-electron   ]                 null
[watch-electron   ]             ],
[watch-electron   ]             "timeout": 0,
[watch-electron   ]             "xsrfCookieName": "XSRF-TOKEN",
[watch-electron   ]             "xsrfHeaderName": "X-XSRF-TOKEN",
[watch-electron   ]             "maxContentLength": -1,
[watch-electron   ]             "maxBodyLength": -1,
[watch-electron   ]             "env": {},
[watch-electron   ]             "headers": {
[watch-electron   ]                 "Accept": "application/json, text/plain, */*",
[watch-electron   ]                 "Content-Type": "application/json",
[watch-electron   ]                 "User-Agent": "axios/0.27.2",
[watch-electron   ]                 "Content-Length": 84
[watch-electron   ]             },
[watch-electron   ]             "method": "post",
[watch-electron   ]             "url": "/user/login/authhash",
[watch-electron   ]             "data": "{\"userName\":\"jamie@remote.it\",\"authHash\":\"861DE10AC9B596DF4C079EA3035862B41DB65DF6\"}"
[watch-electron   ]         },
[watch-electron   ]         "code": "ECONNREFUSED",
[watch-electron   ]         "status": null
[watch-electron   ]     }
[watch-electron   ] }
```

You can do `npm install --legacy-peer-deps` to install and run. Please test and fix. Thank you!

File: backend/src/User.ts [hunk 4]
```diff
@@ -5,9 +5,60 @@ import debug from 'debug'
...
+
+const defaults = {
+  apiURL: 'https://api.remot3.it/apv/v27',
+  successURL: 'https://app.remote.it', // 'https://' + window.location.host
+  taskQueue: 'WeavedTaskQueue',
+  jobQueue: 'WeavedJobQueue',
+}
+
+const instance = setupAxios({ apiURL: API_URL, developerKey: DEVELOPER_KEY })
+instance.interceptors.request.use(
```
Comment (JamieRuderman) [inline backend/src/User.ts:19]: Please see what code is totally needed, and remove anything that is not required.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]

File: backend/src/User.ts [hunk 5]
```diff
@@ -5,10 +5,43 @@ import debug from 'debug'
...
+export function pickBy(object: { [key: string]: any }): any {
+  const obj: { [key: string]: any } = {}
+  for (const key in object) {
+    if (object[key] !== null && object[key] !== false && object[key] !== undefined) {
+      obj[key] = object[key]
+    }
+  }
+  return obj
+}
+
```
Comment (JamieRuderman) [inline backend/src/User.ts:28]: Please see if this can be removed. _Thank you!_

File: backend/src/User.ts [hunk 6]
```diff
@@ -91,3 +144,16 @@ export class User {
...
+export interface IConfig {
+  accessKey?: string
+  apiURL?: string
+  apiKey?: string
+  authHash?: string
+  developerKey?: string
+  successURL?: string
+  taskQueue?: string
+  jobQueue?: string
```
Comment (JamieRuderman) [inline backend/src/User.ts:156]: Please remove unused

File: backend/src/User.ts [hunk 7]
```diff
@@ -38,6 +74,21 @@ export class User {
     EventBus.emit(User.EVENTS.signedIn, this.credentials)
   }
 
+  async authHashLogin(username: string, authhash: string): Promise<any> {
+    return instance
+      .post<IRawUser>('/user/login/authhash', { username, authhash })
+      .then((resp:any) => this.process(resp, username))
+  }
```
Comment (JamieRuderman) [inline backend/src/User.ts:81]: Would it be okay to use async await here?

**Comment by JamieRuderman** [comment]
Please fix authhash login for cli. User.authHash was undefined.

File: backend/src/User.ts (hunk 7 above)
Comment (sonali-remote) [reply backend/src/User.ts]: done, please review

File: backend/src/User.ts [hunk 8]
```diff
@@ -60,6 +65,7 @@ export class User {
       this.username = user.username
       this.authHash = user.authHash
       this.id = user.id
+      this.token = user.token
```
[the remark this answers is not in the document]
Comment (sonali-remote) [reply backend/src/User.ts:99]: removing in signOut also

File: backend/src/User.ts [hunk 9]
```diff
@@ -5,10 +5,31 @@ import debug from 'debug'
 import Logger from './Logger'
 import EventBus from './EventBus'
 import path from 'path'
-import { r3 } from './remote.it'
+import axios from 'axios'
+import { API_URL, DEVELOPER_KEY } from './sharedCopy/constants'
 
-const d = debug('r3:backend:User')
+const defaults = {
+  apiURL: 'https://api.remot3.it/apv/v27',
```
Comment (JamieRuderman) [inline backend/src/User.ts:12]: Please check if this is available in constants file?

File: backend/src/User.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (JamieRuderman) [reply backend/src/User.ts:98]: Please remove and test. Thank you.

**Review by JamieRuderman — APPROVED** [review]

**Pull request merged**